frontend/app/components/redesign/components/LinkTagGenerator.tsx

```diff
@@ -0,0 +1,153 @@
+  const [pointerInput, setPointerInput] = useState('')
+  const [linkTag, setParsedLinkTag] = useState('')
+  const [invalidUrl, setInvalidUrl] = useState(false)
+  const [showCodeBox, setShowCodeBox] = useState(false)
+  const [isCopied, setIsCopied] = useState(false)
+
+  // Function to validate and format the payment pointer URL
+  const isValidPointer = (input: string) => {
+    try {
+      let urlString = input.trim()
+      if (input.charAt(0) === '$') {
+        urlString = input.replace('$', 'https://')
+      }
+
+      const url = new URL(urlString)
+
+      if (url.pathname === '/') {
+        return `${url.origin}.well-known/pay`
+      }
+
+      return url.href
+    } catch (err) {
+      setInvalidUrl(true)
+    }
+  }
+
+  // Just in case the pointer contains special characters
+  const parsePointer = (pointer: string) => {
+    return pointer
+      .replace(/&/g, '&amp;')
+      .replace(/</g, '&lt;')
+      .replace(/>/g, '&gt;')
+      .replace(/"/g, '&quot;')
+      .replace(/'/g, '&apos;')
+  }
+
+  const handleCopyClick = async () => {
+    try {
+      await navigator.clipboard.writeText(
+        `<link rel="monetization" href="${linkTag}" />`
+      )
+      setIsCopied(true)
+      setTimeout(() => setIsCopied(false), 2000)
+    } catch (err) {
+      console.error('Failed to copy text:', err)
+    }
+  }
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault()
+    setInvalidUrl(false)
+    setIsCopied(false)
+
+    const validatedPointer = isValidPointer(pointerInput)
+
+    if (validatedPointer) {
+      setParsedLinkTag(parsePointer(validatedPointer))
+      setShowCodeBox(true)
+    } else {
+      setShowCodeBox(false)
+    }
+  }
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPointerInput(e.target.value)
+    setInvalidUrl(false)
+    setShowCodeBox(false)
+  }
+
+  return (
+    <div className="flex w-full max-w-[800px] p-md flex-col gap-md rounded-sm bg-interface-bg-container">
+      {/* Generator Header */}
+      <div>
+        <Heading5>Link tag generator</Heading5>
+      </div>
+      {/* Input Field */}
+      <div>
+        <label
+          htmlFor="paymentPointer"
+          className={cx(
+            'text-field-helpertext-default font-sans text-xs font-normal leading-xs',
+            invalidUrl && 'text-text-error'
+          )}
+        >
+          Your payment pointer
+        </label>
+        <InputField
+          id="paymentPointer"
+          placeholder="Fill in your payment pointer/wallet address"
+          value={pointerInput}
+          onChange={(e) => handleOnChange(e)}
+          error={
+            invalidUrl
+              ? 'The payment pointer you have entered is not valid or cannot be found'
+              : ''
+          }
+        />
+        {!invalidUrl && (
+          <p className="text-xs text-text-secondary mt-1 sm:hidden">
+            Fill in your payment pointer/wallet address
+          </p>
+        )}
```

Comment:
Instead of adding this, we can make the placeholder font size smaller on smaller screens?